src/compiler/build.c

```diff
@@ -221,19 +263,34 @@ build_prime(build_t* build, const char* rule_name)
 {
 	char    func_name[255];
 	path_t* script_path;
+	const char *ptr_path;
+	duk_int_t rc;
 
 	build->rule_name = strdup(rule_name);
 
 	// process build script
 	emit_op_begin(build, "processing Cellscript.js rule `%s`", rule_name);
 	sprintf(func_name, "$%s", rule_name);
 	script_path = path_rebase(path_new("Cellscript.js"), build->in_path);
-	if (duk_peval_file(build->duktape, path_cstr(script_path)) != 0) {
-		path_free(script_path);
+
+	ptr_path = path_cstr(script_path);
+	push_string_file(build->duktape, ptr_path);  // Source file
+	duk_push_string(build->duktape, ptr_path);   // .fileName property for source
+	path_free(script_path); ptr_path = NULL;
+	rc = duk_pcompile(build->duktape, DUK_COMPILE_EVAL);
+	if (rc != 0) {
+		// XXX: pop?
```

Comment:
No pops necessary on error here.  I didn't bother tracking the value stack in the error paths because if `build_prime()` fails, the Duktape heap is destined to be destroyed anyway.